docs/source/app_developers_guide/docker.rst

```diff
@@ -0,0 +1,416 @@
+  sawtooth-validator-default | writing file: /etc/sawtooth/keys/validator.pub
+  sawtooth-validator-default | creating key directory: /root/.sawtooth/keys
+  sawtooth-validator-default | writing file: /root/.sawtooth/keys/my_key.priv
+  sawtooth-validator-default | writing file: /root/.sawtooth/keys/my_key.pub
+  sawtooth-validator-default | Generated config-genesis.batch
+  sawtooth-validator-default | Processing config-genesis.batch...
+  sawtooth-validator-default | Generating /var/lib/sawtooth/genesis.batch
+
+Open a new terminal so we can connect to the client container:
+
+.. code-block:: console
+
+  % docker exec -it sawtooth-client-default bash
+
+The client container is used to run sawtooth CLI commands, which is the usual
+way to interact with  validators or validator networks.
+
+Your environment is ready for experimenting with Sawtooth. However, any work
+done in this environment will be lost once the container exits. The demo
+compose file provided is useful as a starting point for the creation of your
+own Docker-based development environment. In order to use it for app
+development, you need to take additional steps, such as mounting a host
+directory into the container. See `Docker's documentation
+<https://docs.docker.com/>`_ for details.
+
+
+Resetting The Environment
+-------------------------
+
+If the environment needs to be reset for any reason, it can be returned to
+the default state by logging out of the client container, then pressing
+CTRL-c from the window where you originally ran docker-compose. Once the
+containers have all shut down run 'docker-compose -f sawtooth-default.yaml down'.
+
+.. code-block:: console
+
+  sawtooth-validator-default         | [00:27:56.753 DEBUG    interconnect] message round trip: TP_PROCESS_RESPONSE 0.03986167907714844
+  sawtooth-validator-default         | [00:27:56.756 INFO     chain] on_block_validated: 44ccc3e6(1, S:910b9c23, P:05b2a651)
+  sawtooth-validator-default         | [00:27:56.761 INFO     chain] Chain head updated to: 44ccc3e6(1, S:910b9c23, P:05b2a651)
+  sawtooth-validator-default         | [00:27:56.762 INFO     publisher] Now building on top of block: 44ccc3e6(1, S:910b9c23, P:05b2a651)
+  sawtooth-validator-default         | [00:27:56.763 INFO     chain] Finished block validation of: 44ccc3e6(1, S:910b9c23, P:05b2a651)
+  Gracefully stopping... (press Ctrl+C again to force)
+  Stopping sawtooth-tp_xo_python-default ... done
+  Stopping sawtooth-tp_config-default ... done
+  Stopping sawtooth-client-default... done
+  Stopping sawtooth-rest_api-default ... done
+  Stopping sawtooth-tp_intkey_python-default ... done
+  Stopping sawtooth-validator-default ... done
+
+  % docker-compose -f sawtooth-default.yaml down
+
+
+
+.. note::
+
+  Genesis block, validator and transaction processor startup are handled for you
+  if you're using the Docker workflow. If you want a better understanding of
+  this process, see :doc:`ubuntu`.
```

Comment:
This would make more sense right after the docker-compose up command is run and could be as simple as "To learn more about the startup process, see: ..."